### ISSUE TITLE
Warm-up stage training

### ISSUE THREAD

**Issue by saolaisa02** [microsoft/ProDA#27]
Can you share how to train warm-up stage?
I use this script to train warm-up stage but the result is quite fail and can't get 43.3 mIoU like pre-trained.
`python train.py --name gta2citylabv2_warmup --stage warm_up --student_init simclr --no_resume --lr 0.0001`

**Comment by panzhang0212** [comment]
The warmup is the method of [AdaptSegNet](https://github.com/wasidennis/AdaptSegNet), and you can read more from their paper and code. Our implementation is a little bit worse than the official version and the training of AdaptSegNet is not stable, our implementation params: --warm_up --freeze_bn --gan LS --lr 2.5e-4 --adv 0.01 --no_resume, and the performance larger than 40 mIoU is ok, In the paper, the warmup is 41.6mIoU

**Comment by panzhang0212** [comment]
We borrow warmup from [CAG_UDA](https://github.com/RogerZhangzz/CAG_UDA) and [Uncertainty](https://github.com/layumi/Seg-Uncertainty). Maybe the warmups of their two work are more efficient, and you can learn more warmup from these two work.

**Comment by manhminno** [comment]
Hello @panzhang0212 ,
As you say, the performance larger than 40 mIoU is ok, in your paper, the warmup is 41.6mIoU. But the pre-trained you have published, the mIoU reaches 43.3, so how do you train to achieve 43.3mIoU, where did this result come from?

**Comment by saolaisa02** [comment]
@panzhang0212 why why didn't you answer mino's question? 
If warm-up stage only gets 41.6, stage 1 can't get 53.7, eventually the result will be bad on gta5->cityscapes (can't get 57.5 as you announced in the article). Where did you get the pretrained of warm-up result 43.3? Have you cheated?

**Comment by panzhang0104** [comment]
Sorry for the delay due to busy work. Below is the training log
[run_2020_11_23_11_18_18.log](https://github.com/microsoft/ProDA/files/7351367/run_2020_11_23_11_18_18.log)

the 41.6 in the paper is our warm-up result (the log is below). After cleaning the code for release, I retrain all stages to ensure the correctness of the code, the log is above, warp-up performs unstable, but the final results are similar.
[run_2020_07_27_05_03_11.log](https://github.com/microsoft/ProDA/files/7351380/run_2020_07_27_05_03_11.log)

If you are interested in the warp-up stage, I will merge the warm-up training with all details into the repo in the next few weeks.

below are all training log:
https://github.com/microsoft/ProDA/issues/1#issuecomment-818553594
> Hello @panzhang0212 , As you say, the performance larger than 40 mIoU is ok, in your paper, the warmup is 41.6mIoU. But the pre-trained you have published, the mIoU reaches 43.3, so how do you train to achieve 43.3mIoU, where did this result come from?

**Comment by panzhang0104** [comment]
As shown above, 41.6 is from our uncleaned code, but unclean code and clean code perform similarly.  Do you need to try our 41.6 warm-up model? I can share it with you.  And the warm-up method is from https://github.com/wasidennis/AdaptSegNet, the repo could get 44.1 with LSGAN. And we merge the method as warm-up, the performance is 41.6 and 43.3 in two version codes. Due to unstable adversarial training, I think the performance is reasonable.


> @panzhang0212 why why didn't you answer mino's question? If warm-up stage only gets 41.6, stage 1 can't get 53.7, eventually the result will be bad on gta5->cityscapes (can't get 57.5 as you announced in the article). Where did you get the pretrained of warm-up result 43.3? Have you cheated?

**Comment by shahaf1313** [comment]
Hey :) 
@panzhang0104 could you please share the exact command line that you used in order to train the warmup model? 
From the discussion above I didn't realize what are the parameters for the warmup stage.
I tried:
`python ./train.py --stage warmup --name gta2citylabv2_warmup --S_pseudo_src 2 `
And it didn't work.
Thanks!

**Comment by yasserben** [comment]
Hi @panzhang0104 , can you please provide the training logs of the remaining training stages (stage 1, 2 and 3) please ?
Thank you very much !

**Comment by panzhang0104** [comment]
@yasserben 
https://github.com/microsoft/ProDA/issues/27#issuecomment-944042633

**Comment by panzhang0104** [comment]
@shahaf1313 
https://github.com/microsoft/ProDA/issues/42#issuecomment-1085366024
